nni/algorithms/compression/v2/pytorch/pruning/tools/sparsity_allocator.py

```diff
@@ -178,26 +178,42 @@ def generate_sparsity(self, metrics: Dict) -> Dict[str, Dict[str, Tensor]]:
         self._get_dependency()
         masks = {}
         grouped_metrics = {}
+        grouped_names = set()
+        # combine metrics with channel dependence
         for idx, names in enumerate(self.channel_depen):
             grouped_metric = {name: metrics[name] for name in names if name in metrics}
+            grouped_names.update(set(grouped_metric.keys()))
```

Comment:
The parameter type is iterable. No need to convert.
https://github.com/python/typeshed/blob/master/stdlib/builtins.pyi#L980

Reply:
wow, got it, fixed